lib/oban/notifier.ex

```diff
@@ -130,10 +128,19 @@ defmodule Oban.Notifier do
       Oban.Notifier.listen([:gossip, :insert, :signal])
   """
   @spec listen(GenServer.server(), channels :: list(channel())) :: :ok
-  def listen(server \\ __MODULE__, channels) when is_server(server) and is_list(channels) do
+  def listen(server \\ __MODULE__, channels)
```

Comment:
This needs to use the oban instance name, which defaults to `Oban`.

```suggestion
  def listen(oban_name \\ Oban, channels)
```